### PR TITLE
Bug 1231515 - Changed opening of domain to full url for default top sites

### PR DESCRIPTION
**Pull request by sleroux**

### CODE REVIEW

File: Client/Frontend/Home/TopSitesPanel.swift [hunk 1]
```diff
@@ -260,7 +260,7 @@ extension TopSitesPanel: UICollectionViewDelegate {
         if let site = dataSource[indexPath.item] {
             // We're gonna call Top Sites bookmarks for now.
             let visitType = VisitType.Bookmark
-            let destination = NSURL(string: site.url)?.domainURL() ?? NSURL(string: "about:blank")!
+            let destination = NSURL(string: site.url) ?? NSURL(string: "about:blank")!
```
Comment (rnewman) [inline Client/Frontend/Home/TopSitesPanel.swift]: This isn't right: it'll make *all* Top Sites load whichever full URL contributed the most to the tile.

Comment (rnewman) [inline Client/Frontend/Home/TopSitesPanel.swift]: Might be time to act on my note elsewhere in this file:

```
        // The obvious solution here and in collectionView:didSelectItemAtIndexPath: is for the cursor
        // to return domain sites, not history sites -- that is, with the right icon, title, and URL --
        // and for this code to just use what it gets.
```

File: Client/Frontend/Home/TopSitesPanel.swift [hunk 2]
```diff
@@ -260,7 +260,14 @@ extension TopSitesPanel: UICollectionViewDelegate {
         if let site = dataSource[indexPath.item] {
             // We're gonna call Top Sites bookmarks for now.
             let visitType = VisitType.Bookmark
-            let destination = NSURL(string: site.url)?.domainURL() ?? NSURL(string: "about:blank")!
+
+            let destination: NSURL
+            if site is SuggestedSite {
```
Comment (rnewman) [inline Client/Frontend/Home/TopSitesPanel.swift]:
```
extension Site {
    tileURL: NSURL {
        return NSURL(string: site.url)?.domainURL() ?? NSURL(string: "about:blank")!
    }
}
```

and override in `SuggestedSite`. Or just dump it straight into the class definitions.

Then here:

```
let destination = site.tileURL
```

Comment (sleroux) [inline Client/Frontend/Home/TopSitesPanel.swift]: Sadly extension functions/vars can't be overriden yet :( I'll add it to the class and override.

**Comment by sleroux** [comment]
@rnewman Updated PR with better approach. Thanks Richard!

**Comment by rnewman** [comment]
Ship it!

**Pull request merged**